### PR TITLE
WDIO5 target regions

### PR DESCRIPTION
**Pull request by danielschwartz85**
* Added WDIO 5 target regions support for VG eyes
* Fix xpath script in core
* Fix EyesWebElements regsions
* Moved persisted regions calculation to checkSettings (will write comments about this)  
* refactor region tests to use real driver

### CODE REVIEW

File: packages/eyes-sdk-core/lib/fluent/CheckSettings.js [hunk 1]
```diff
@@ -346,6 +346,10 @@ class CheckSettings {
     throw new TypeError('Method called with argument of unknown type!')
   }
 
+  _getTargetType() {
```
Comment (danielschwartz85) [inline packages/eyes-sdk-core/lib/fluent/CheckSettings.js]: wdio5 checksettings overrides this

File: packages/eyes-sdk-core/lib/fluent/CheckSettings.js [hunk 2]
```diff
@@ -560,6 +564,37 @@ class CheckSettings {
   toString() {
     return `${this.constructor.name} ${GeneralUtils.toString(this)}`
   }
+
```
Comment (danielschwartz85) [inline packages/eyes-sdk-core/lib/fluent/CheckSettings.js]: I think other SDKs would also use _getPersistedregions(webDriver) method so moved it now to checkSettings

File: packages/eyes-webdriverio-5/src/EyesVisualGrid.js [hunk 3]
```diff
@@ -259,46 +254,46 @@ class EyesVisualGrid extends EyesBase {
...
-    const [ignore, floating, strict, layout, content, accessibility] = await Promise.all([
-      this._persistRegions(checkSettings.getIgnoreRegions()),
-      this._persistRegions(checkSettings.getFloatingRegions()),
-      this._persistRegions(checkSettings.getStrictRegions()),
-      this._persistRegions(checkSettings.getLayoutRegions()),
-      this._persistRegions(checkSettings.getContentRegions()),
-      this._persistRegions(checkSettings.getAccessibilityRegions()),
+    const [config, {region, selector}] = await Promise.all([
+      checkSettings.toCheckWindowConfiguration(this._driver),
+      this._getTargetConfiguration(checkSettings),
```
Comment (danielschwartz85) [inline packages/eyes-webdriverio-5/src/EyesVisualGrid.js]: notice getTargetConfiguration (the persisted target regions are still in SDK responsibility and not core, since SDK checksettings has properties regarding this which core/checkSettings doesn't know about (some of this can maybe also be moved to core in the future..)

File: packages/eyes-webdriverio-5/src/fluent/AccessibilityRegionByElement.js [hunk 4]
```diff
@@ -46,7 +46,7 @@ class AccessibilityRegionByElement extends GetAccessibilityRegion {
   }
 
   async toPersistedRegions(driver) {
-    const xpath = await EyesJsBrowserUtils.getElementXpath(driver, this._element)
+    const xpath = await EyesJsBrowserUtils.getElementXpath(driver, this._element.element)
```
Comment (danielschwartz85) [inline packages/eyes-webdriverio-5/src/fluent/AccessibilityRegionByElement.js]: this is unrelated to refactor - just a fix to make regions of EyesWebElement work

File: packages/eyes-webdriverio-5/src/EyesVisualGrid.js [hunk 5]
```diff
@@ -259,46 +254,46 @@ class EyesVisualGrid extends EyesBase {
...
-        ? checkSettings.getMatchLevel()
-        : this.getMatchLevel(),
       source,
     })
   }
 
-  async _persistRegions(regions) {
-    const persisted = await Promise.all(regions.map(r => r.toPersistedRegions(this._driver)))
-    return [].concat(...persisted)
+  async _getTargetConfiguration(checkSettings) {
```
Comment (kyryloonufriiev) [inline packages/eyes-webdriverio-5/src/EyesVisualGrid.js]: Why `getTargetProvider` and `getTargetRegion` are separate functions? As I understand there couldn't be both at the same time.

Comment (danielschwartz85) [reply packages/eyes-webdriverio-5/src/EyesVisualGrid.js]: It seems that getTargetRegion is sdk-core and getTargetProvider is SDKs.
They are used in different ways in Classic - when user want's element as target is a different code path from use who want's region (src/EyesWDIO.js:362).
Also the VG needs to differentiate, 

So we could say let's make a function that returns either ```{selector}``` or ```{region}``` and then we could tell the difference but perhaps even saving them in the same property on checkSettings is the right thing.. anyway here also I preferred not handle this since it's a bit of straying away..

**Comment by danielschwartz85** [comment]
@kyryloonufriiev 
* removing unused ```_prepareRegions```
* regrading ```getSelector```:
 Many regions have SelectorByLocator and SelectorByElement and so they use it's ```getSelector``` method, but many regions don't have it (I thought about just using these classes as Util functions and adding them to regions that don't use them - so we can call getSelector).
Also note that ```getSelector``` is not really returning a real selector (since they generate a selector set it in the DOM and return the generated selector - which then the VG doesn't see).
We could maybe override this functionality and calculate xpath etc..
So not sure how to go about this, but generally I think this issue straying from this PR.

Note also that regarding ```getSelector``` we can probably do more to have a cleaner way to persist all the regions - this is very related to upcoming code changes we want to do which may have a bigger scope :)
